pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -271,12 +293,61 @@ func (m *ManifestWorkController) applyOneManifest(
 	// TODO we should check the certain error.
 	// Use dynamic client when scheme cannot decode manifest or typed client cannot handle the object
 	if isDecodeError(result.Error) || isUnhandledError(result.Error) || isUnsupportedError(result.Error) {
-		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, unstructuredObj, gvr, recorder)
+		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, required, gvr, recorder)
 	}
 
 	return result
 }
 
+func (m *ManifestWorkController) serverSideApply(
+	ctx context.Context,
+	gvr schema.GroupVersionResource,
+	required *unstructured.Unstructured,
+	config *workapiv1.ServerSideApplyConfig,
+	recorder events.Recorder) (*unstructured.Unstructured, bool, error) {
+	force := false
+	fieldManager := "work-agent"
+
+	if config != nil {
+		force = config.Force
+		if len(config.FieldManager) > 0 {
+			fieldManager = config.FieldManager
+		}
+	}
+
+	existing, err := m.spokeDynamicClient.
```

Comment:
under SSA mode, we are not supposed to do any client-side get any more. all the owner-ref and label/annotation merging should be done at the server-side. can you elaborate the purpose of this?